### DISCUSSION TITLE
[req] Support for defining a humidistat to control a humidifier

### DISCUSSION THREAD

**Discussion by dekimsey** [jmcollin78/versatile_thermostat#750] [Ideas]
First off, props to the project this was a really easy integration to use. The config flow was really well done! Very cool!

tl;dr It would be neat if the project could support alternative climate devices to create a humidistat and not just temperature. It may be out of scope, but it would be great! 

**Problem:** In my case, I needed a way to control a dumb humidifier because with winter let's just say my sinus's aren't well adapted to this 36% humidity at night. But if I leave it on, the humidity is usually too high by morning. A set-point type climate control is exactly the behavior I want, just for humidity not temperature.

**Solution:** versatile_thermostat's `over_switch` thermostat with inputs from my a humidity sensor in the room.

<details><summary>Implementation Details (switch, sensor, humidifier)</summary>

I was able to do this by simply configuring an `over_switch` to:
- switch.humidifier
- sensor.bedroom_humidity
- sensor.bedroom_last_seen
- sensor.xxx_outdoor_humidity
- sensor.bedroom_motion_detection
- sensor.phone_presence

Corresponding to:
- dumb humidifier (Honeywell Cool Moisture Humidifier, HCM-350. This has 3 fan modes, I just leave it on high)
- smart switch (Zigbee, Sengled E1C-NB6)
- a humidity sensor (Z-Wave, AEON Labs MultiSensor 6 ZW100)
- a weather sensor for outdoor humidity (NOAA)
Enabled Optional:
- a motion detector (same multisensor device)
- presence sensor (me, cellphone's HA app)

</details>

It works well enough, but it sure does look a bit silly.

![image](https://github.com/user-attachments/assets/e9567412-3500-4159-8525-5b06149aa2e4 "the device controls, all set the set points all actually being values for humidity %")
![image](https://github.com/user-attachments/assets/44d440a7-0278-46c1-8995-bda72bfe4649 "reports in °F, but heat means increase so it works")

Related:
- [Add humidity sensor/attribute to climate entity #633](https://github.com/jmcollin78/versatile_thermostat/discussions/633)

PS: After setting all of this up last night and being very pleased with the effect this morning (sinus's hurt significantly less). I learned there is native support in HA via [generic_hygrostat](https://www.home-assistant.io/integrations/generic_hygrostat/), so I'll try that too. But I suspect the algorithm here is much better, and I particularly like the rate sensor anyway.

**Comment by jmcollin78** [comment]
Hello,

Out of the box, you can control the on/off with a switch. But all algorithm are made for temperature. Not sure how they will behave with humidity. And like you notice it, all displays are made for °.

**Reply by dekimsey** [reply]
Yep, agreed! Gotta say, so far it works better than the built in integration at keeping a setpoint ;)
